### ISSUE TITLE
Does 'convert_data2vec_text_original_pytorch_checkpoint_to_pytorch.py' transfer data2vec-text model's parameter well?

### ISSUE THREAD

**Issue by qwer4107** [huggingface/transformers#18277]
### System Info

```shell
- `transformers` version: 4.21.0.dev0
- Platform: Linux-3.10.0-1160.42.2.el7.x86_64-x86_64-with-glibc2.17
- Python version: 3.8.13
- Huggingface_hub version: 0.8.1
- PyTorch version (GPU?): 1.10.1+cu111 (False)
```

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I am trying to convert own pretrained data2vec-text to Hugginface form, but it looks there are some parameter mismatches on code. 

In my opinion there seems some troubleshooting parts are included in 'convert_data2vec_text_original_pytorch_checkpoint_to_pytorch.py'

1. import error
I pretrained my model with fairseq code, so that I should import it with fairseq one (doesn't work with suggested code with import Data2vecModel from transformers). I tweak this part to load my model with 'Data2vecTextModel' from fairseq, then it works.

2. 'lm_head'?
IIRC, data2vec's pretraining procedure updates model parameters with embedding vector(the Avg. of top k output of Transformer Network). So, there should be no 'lm_head', I guess. As I guess, my pre-trained model only has 'regression_head' layer not 'lm_head' layer. However, Data2vecForMaskedLM's architecture (ofcoursely?) tries to transfer lm_head's parameters, which i don't have. To solve this problem, I tweak the code to import Data2vecModel from transformers to transfer headless parameters. 

![스크린샷 2022-07-25 오후 4 09 54](https://user-images.githubusercontent.com/71030815/180718595-e31a0d8a-7eef-4e25-86d1-7869fa9a5eb4.png)

As you can see, the upper on image is Huggingface's Data2vecTextModel architecture, and the below is Fairseq's one(my own pretrained model from fairseq). 

Can you guide me if there's something i misunderstood the module? 
There's no need to transfer 'regression_head', and it's fine to leave blank with 'Pooler'?

### Expected behavior

```shell

```

### Checklist

- [X] I have read the migration guide in the readme. ([pytorch-transformers](https://github.com/huggingface/transformers#migrating-from-pytorch-transformers-to-transformers); [pytorch-pretrained-bert](https://github.com/huggingface/transformers#migrating-from-pytorch-pretrained-bert-to-transformers))
- [X] I checked if a related official extension example runs on my machine.

**Comment by LysandreJik** [comment]
cc @patrickvonplaten

**Comment by qwer4107** [comment]
I solved this issue myself by tweaking the codes, leaving pooler parameters blank but others to be transferred. I import 

In my opinion, however, there's no MLM task in data2vec Text model's pretraining procedure, so imported model should be 'Data2VecTextModel' not 'Data2VecTextForMaskedLM'. 

For data2vec, MaskedLM is one of the downstream tasks, not pretraining way. They only require the K top of outputs(the latent representation) of Transformers Network for distillation and parameter updates, which is different from BERT-like way. 

Anyway, no pretraining way is now on service in Hugginface, so that there's no need to transfer 'regression_head' if I understand correctly. 

I'd like to listen other opinions because I may misunderstood sth. 

Thanks

**Comment by patrickvonplaten** [comment]
Hey @qwer4107,

Good analysis. We indeed didn't find the time yet to port the pretraining capabilities of the model (happy to help in adding such functionality if you'd be interested though!)

The reason why we added the class nevertheless is because we want to be sure to be able to add pretraining code in the future without breaking backwards compatibility. You're right that for now there is no need to convert the regression_head!
